pages/[slug].tsx

```diff
@@ -0,0 +1,128 @@
+    </ContentPageLayout>
+  )
+}
+
+export const getStaticProps:GetStaticProps = async ({ params }) => {
+  const pageFilePath = path.join(contentFilePath, `${params.slug}.mdx`)
+  const source = fs.readFileSync(pageFilePath)
+  const { content, data } = matter(source)
+  // get all the siblings for the section and pass them to
+  // the section navigation
+  const pageSection = data?.section
+  const allPagesMeta = []
+  fs.readdirSync(contentFilePath, 'utf8')
+    .filter((path) => /\.mdx?$/.test(path))
+    .forEach((file) => {
+      const pageSource = matter(fs.readFileSync(path.join(contentFilePath, file), 'utf8'))
+      const pageMeta = pageSource.data
+      pageMeta.path = `/${file.replace(/\.mdx$/, '')}/`
+      allPagesMeta.push(pageMeta)
+    })
+
+  const siblings = allPagesMeta.filter(item => item.section === pageSection)
+
+  // add table of content to pages
+  const addTocToPage = () => {
+    // don't generate TOC on some pages
+    if (!['404', 'cookies', 'sign-in'].some(v => data.path.includes(v))) {
+      return [
+        rehypeSlug,
+        [toc, {
+          headings: ['h2', 'h3', 'h4'],
+          cssClasses: {
+            toc: 'app-toc',
+            list: 'app-toc__list',
+            listItem: 'app-toc__list-item',
+            link: 'app-toc__link'
+          },
+          customizeTOC: function (toc) {
+            if (toc.type === 'element' && toc.tagName === 'nav') {
+              toc.properties.role = 'navigation'
+              toc.properties['aria-label'] = 'Sections on this page'
+            }
+            if (toc.children[0].children.length === 0) {
+              return false
+            } else {
+              toc.children.unshift({
+                type: 'element',
+                tagName: 'h2',
+                children: [
+                  {
+                    type: 'text',
+                    value: 'Contents'
+                  }
+                ],
+                properties: {
+                  className: 'app-toc__heading'
+                }
+              })
+              return toc
+            }
+          }
+        }]
+      ]
+    } else return []
+  }
+
+  const mdxSource = await serialize(content, {
+    // Optionally pass remark/rehype plugins
+    mdxOptions: {
+      remarkPlugins: [],
+      rehypePlugins: addTocToPage()
+    },
+    scope: data
+  })
+
+  return {
+    props: {
+      source: mdxSource,
+      frontMatter: data,
+      siblings: siblings
+    }
+  }
+}
+
+export const getStaticPaths:GetStaticPaths = async () => {
+  const paths = fs
+    .readdirSync(contentFilePath)
+    .filter((path) => /\.mdx?$/.test(path))
+    // Remove file extensions for page paths
+    .map((path) => path.replace(/\.mdx?$/, ''))
+    // exclude homepage as it has its own template
+    .filter((path) => !path.includes('index'))
+    // Map the path into the static paths object required by Next.js
+    .map((slug) => ({ params: { slug } }))
```

Comment:
Big fan of this chain pattern - however I think the need to comment them might suggest that these functions could do with names?
How about something like:

```js
const remove_mdx = path => path.replace(/\.mdx?$/, '')
const paths_without_index = path => !path.includes('index')
const paths_object = slug => ({ params: { slug } })
```

Reply:
hmm these are not anonymous functions. 
it's map and filter methods on the Array class and `path => !path.includes('index')` for example is a test 

not sure it makes sense to create another set of named functions to return matching results. do you have any good examples of this?

Reply:
They are equivalent, right? https://www.w3schools.com/Js/js_arrow_function.asp

Reply:
Cool!